src/proj.rs

```diff
@@ -1286,6 +1288,142 @@ impl Proj {
+                    "ALLOW_ELLIPSOIDAL_HEIGHT_AS_VERTICAL_CRS={}",
+                    if allow_ellipsoidal_height_as_vertical_crs {
+                        "YES"
+                    } else {
+                        "NO"
+                    }
+                ))?);
+            }
+
+            if let Some(allow_linunit_node) = options.allow_linunit_node {
+                opts.push(CString::new(format!(
+                    "ALLOW_LINUNIT_NODE={}",
+                    if allow_linunit_node { "YES" } else { "NO" }
+                ))?);
+            }
+
+            if opts.is_empty() {
+                None
+            } else {
+                Some(opts)
+            }
+        } else {
+            None
+        };
+
+        let opts_ptrs = options_str
+            .as_ref()
+            .map(|o| o.iter().map(|cs| cs.as_ptr()).collect::<Vec<_>>());
+
+        let wkt_type = match version.unwrap_or(WktVersion::Wkt2_2019) {
+            WktVersion::Wkt2_2015 => PJ_WKT_TYPE_PJ_WKT2_2015,
+            WktVersion::Wkt2_2015_Simplified => PJ_WKT_TYPE_PJ_WKT2_2015_SIMPLIFIED,
+            WktVersion::Wkt2_2019 => PJ_WKT_TYPE_PJ_WKT2_2019,
+            WktVersion::Wkt1_Gdal => PJ_WKT_TYPE_PJ_WKT1_GDAL,
+            WktVersion::Wkt1_Esri => PJ_WKT_TYPE_PJ_WKT1_ESRI,
+        };
+
+        unsafe {
+            let wkt = proj_as_wkt(
+                self.ctx,
+                self.c_proj,
+                wkt_type,
+                opts_ptrs
+                    .as_ref()
+                    .map(|c| c.as_ptr())
+                    .unwrap_or(ptr::null()),
+            );
+
+            Ok(_string(wkt)?)
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct WktOptions {
+    /// Defaults to YES, except for WKT1_ESRI
+    multiline: Option<bool>,
+
+    /// Defaults to 4 (when multiline output is on).
+    indentation_width: Option<usize>,
+
+    /// In AUTO mode, axis will be output for WKT2 variants, for WKT1_GDAL for ProjectedCRS with
+    /// easting/northing ordering (otherwise stripped), but not for WKT1_ESRI. Setting to YES will
+    /// output them unconditionally, and to NO will omit them unconditionally.
+    output_axis: Option<WktOutputAxis>,
+
+    /// Default is YES. If NO, a Geographic 3D CRS can be for example exported as WKT1_GDAL with
+    /// 3 axes, whereas this is normally not allowed.
+    strict: Option<bool>,
+
+    /// Default is NO. If set to YES and type == PJ_WKT1_GDAL, a Geographic 3D CRS or a Projected 3D
+    /// CRS will be exported as a compound CRS whose vertical part represents an ellipsoidal height
+    /// (for example for use with LAS 1.4 WKT1).
+    allow_ellipsoidal_height_as_vertical_crs: Option<bool>,
+
+    /// Default is YES starting with PROJ 9.1. Only taken into account with type == PJ_WKT1_ESRI on a Geographic 3D CRS.
+    allow_linunit_node: Option<bool>,
+}
+
+pub enum WktOutputAxis {
+    Auto,
+    Yes,
+    No,
+}
+
+#[allow(non_camel_case_types)]
+pub enum WktVersion {
```

Comment:
It seems like there are some missing entries here: https://proj.org/en/stable/development/reference/datatypes.html#c.PJ_WKT_TYPE

I'm not familiar enough to know if these will be missed.

Reply:
Sounds reasonable to me. 👍